### PR TITLE
Update CoreFx and core-setup dependencies manually

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   <ProductDependencies>
-    <Dependency Name="Microsoft.CSharp" Version="4.6.0-preview4.19160.8">
+    <Dependency Name="Microsoft.CSharp" Version="4.6.0-preview4.19162.2" CoherentParentDependency="Microsoft.NETCore.App">
       <Uri>https://github.com/dotnet/corefx</Uri>
-      <Sha>6d7126e0a329db795930ea17e84369eacb7bae5a</Sha>
+      <Sha>7c36280bd83b85bf5b3f229d3c1a293f00ef59dd</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.PlatformAbstractions" Version="3.0.0-preview4-27511-16">
+    <Dependency Name="Microsoft.DotNet.PlatformAbstractions" Version="3.0.0-preview4-27512-03" CoherentParentDependency="Microsoft.NETCore.App">
       <Uri>https://github.com/dotnet/core-setup</Uri>
```
Comment (dougbu) [inline eng/Version.Details.xml]: @mmitche you mentioned `CoherentParentDependency` wasn't needed for dotnet/core-setup dependencies. Does it hurt?

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -45,25 +45,30 @@
...
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.Platforms" Version="3.0.0-preview4.19160.8">
+    <Dependency Name="Microsoft.NETCore.Platforms" Version="3.0.0-preview4.19162.2" CoherentParentDependency="Microsoft.NETCore.App">
       <Uri>https://github.com/dotnet/corefx</Uri>
-      <Sha>6d7126e0a329db795930ea17e84369eacb7bae5a</Sha>
+      <Sha>7c36280bd83b85bf5b3f229d3c1a293f00ef59dd</Sha>
     </Dependency>
-    <Dependency Name="System.Collections.Immutable" Version="1.6.0-preview4.19160.8">
+    <!-- This dependency is not used directly. It exists to ground the CoreFx dependencies. -->
+    <Dependency Name="Microsoft.NETCore.App" Version="3.0.0-preview4-27512-03">
```
Comment (dougbu) [inline eng/Version.Details.xml]: @bricelam this is just a draft but wanted your input on adding this "fake" `<Dependency />`.

**Comment by dougbu** [comment]
Oh good, the checks passed. No longer in draft status.

Any objections @mmitche or @bricelam ?

File: eng/Version.Details.xml (hunk 2 above)
Comment (bricelam) [reply eng/Version.Details.xml]: Looks good. The comment can probably be removed. Its purpose is pretty obvious from `CoherentParentDependency="Microsoft.NETCore.App"`.

File: eng/Version.Details.xml (hunk 1 above)
Comment (dougbu) [reply eng/Version.Details.xml]: Ping @mmitche (I'm ready to merge this and don't want later to hear I shouldn't have added `CoherentParentDependency` here or for `Microsoft.Extensions.DependencyModel`.)

**Pull request merged**